### PR TITLE
feature: add styling for non-whitelisted balances

### PR DESCRIPTION
**Pull request by StefChatz**
photo_2024-09-10 17 52 48
photo_2024-09-10 17 53 03

Enhanced AccountBalancesTable to visually distinguish and sort non-whitelisted assets, improving user understanding of collateral status in the account balances view using the tooltip of type info component.

### CODE REVIEW

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 1]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+
+    const wrapWithTooltip = (content: React.ReactNode, denom: string) => {
+      if (isWhitelisted(denom)) return content
+      return (
+        <Tooltip
+          type='info'
+          content="This asset is not whitelisted and doesn't count as collateral"
+        >
+          <div className='cursor-help'>{content}</div>
+        </Tooltip>
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:43]:
```suggestion
        <Tooltip
          type='info'
          content="This asset is not whitelisted and doesn't count as collateral"
          className='cursor-help'
        >
          {content}
        </Tooltip>
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 2]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
   const { computeLiquidationPrice } = useHealthComputer(updatedAccount ?? account)
 
   return useMemo<ColumnDef<AccountBalanceRow>[]>(() => {
+    const isWhitelisted = (denom: string) => whitelistedAssets.find(byDenom(denom)) !== undefined
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:33]:
```suggestion
    const asset = useAsset(row.original.denom)
    const isWhitelisted = !!asset?.isWhitelisted
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 3]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+
+    const wrapWithTooltip = (content: React.ReactNode, denom: string) => {
+      if (isWhitelisted(denom)) return content
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:36]:
```suggestion
    const wrapWithTooltip = (content: React.ReactNode, isWhitelisted: boolean) => {
      if (isWhitelisted) return content
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 4]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+      )
+    }
+
+    const applyRowStyling = (children: React.ReactNode, denom: string) => (
+      <div
+        className={`flex items-center ${!isWhitelisted(denom) ? 'opacity-80 hover:opacity-100' : ''}`}
+      >
+        {children}
+      </div>
+    )
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:53]:
```suggestion
    const applyRowStyling = (children: React.ReactNode, isWhitelisted: boolean) => (
      <div
        className={classNames('flex items-center', !isWhitelisted && 'opacity-80 hover:opacity-100')}
      >
        {children}
      </div>
    )
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 5]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
         ...ASSET_META,
-        cell: ({ row }) => <Asset type={row.original.type} symbol={row.original.symbol} />,
+        cell: ({ row }) =>
+          wrapWithTooltip(
+            applyRowStyling(
+              <Asset type={row.original.type} symbol={row.original.symbol} />,
+              row.original.denom,
+            ),
+            row.original.denom,
+          ),
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:65]:
```suggestion
        cell: ({ row }) =>
          wrapWithTooltip(
            applyRowStyling(
              <Asset type={row.original.type} symbol={row.original.symbol} />,
              isWhitelisted,
            ),
            isWhitelisted,
          ),
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 6]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+            applyRowStyling(
+              <Value
+                amountChange={row.original.amountChange}
+                value={row.original.value}
+                type={row.original.type}
+              />,
+              row.original.denom,
+            ),
+            row.original.denom,
+          ),
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:80]:
```suggestion
          wrapWithTooltip(
            applyRowStyling(
              <Value
                amountChange={row.original.amountChange}
                value={row.original.value}
                type={row.original.type}
              />,
              isWhitelisted,
            ),
            isWhitelisted,
          ),
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 7]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+              <Size
+                size={row.original.size}
+                amountChange={row.original.amountChange}
+                denom={row.original.denom}
+                type={row.original.type}
+              />,
+              row.original.denom,
+            ),
+            row.original.denom,
+          ),
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:97]:
```suggestion
              <Size
                size={row.original.size}
                amountChange={row.original.amountChange}
                denom={row.original.denom}
                type={row.original.type}
              />,
              isWhitelisted,
            ),
            isWhitelisted,
          ),
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 8]
```diff
@@ -29,45 +30,89 @@ export default function useAccountBalancesColumns(
...
+                  applyRowStyling(
+                    <Price
+                      type={row.original.type}
+                      amount={row.original.amount.toNumber()}
+                      denom={row.original.denom}
+                    />,
+                    row.original.denom,
+                  ),
+                  row.original.denom,
+                ),
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:115]:
```suggestion
                wrapWithTooltip(
                  applyRowStyling(
                    <Price
                      type={row.original.type}
                      amount={row.original.amount.toNumber()}
                      denom={row.original.denom}
                    />,
                    isWhitelisted,
                  ),
                  isWhitelisted,
                ),
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 9]
```diff
@@ -77,29 +122,39 @@ export default function useAccountBalancesColumns(
...
+                      denom={row.original.denom}
+                      computeLiquidationPrice={computeLiquidationPrice}
+                      type={row.original.type}
+                      amount={row.original.amount.toNumber()}
+                      account={updatedAccount ?? account}
+                      isWhitelisted={isWhitelisted(row.original.denom)}
+                    />,
+                    row.original.denom,
+                  ),
+                  row.original.denom,
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:138]:
```suggestion
                    <LiqPrice
                      denom={row.original.denom}
                      computeLiquidationPrice={computeLiquidationPrice}
                      type={row.original.type}
                      amount={row.original.amount.toNumber()}
                      account={updatedAccount ?? account}
                      isWhitelisted={isWhitelisted}
                    />,
                    isWhitelisted,
                  ),
                  isWhitelisted,
```

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 10]
```diff
@@ -77,29 +122,39 @@ export default function useAccountBalancesColumns(
...
+              <Apy
+                apy={row.original.apy}
+                markets={markets}
+                denom={row.original.denom}
+                type={row.original.type}
+              />,
+              row.original.denom,
+            ),
+            row.original.denom,
+          ),
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:157]:
```suggestion
        cell: ({ row }) =>
          wrapWithTooltip(
            applyRowStyling(
              <Apy
                apy={row.original.apy}
                markets={markets}
                denom={row.original.denom}
                type={row.original.type}
              />,
              isWhitelisted,
            ),
            isWhitelisted,
          ),
```

File: src/components/common/Table/Row.tsx [hunk 11]
```diff
@@ -43,6 +46,7 @@ export default function Row<T>(props: Props<T>) {
           canExpand && row.getIsExpanded()
             ? 'is-expanded border-t gradient-header'
             : 'hover:bg-white/5',
+          !isWhitelisted && 'opacity-50 hover:opacity-100',
```
Comment (linkielink) [inline src/components/common/Table/Row.tsx:49]: What is `applyRowStyling` needed for if you also style the row here?

**Review by linkielink — CHANGES_REQUESTED** [review]
I think a hook `useWhitelistedAsset(denom: string):boolean` would be a good addition. It could work like `useAsset(denom: string):Asset`.

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx (hunk 1 above)
Comment (StefChatz) [reply src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:43]: the info type already adds cursor-help so no point adding it

Comment (linkielink) [reply src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:44]: You can remove the wrapper `<div className='cursor-help'>{content}</div>` then

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx (hunk 2 above)
Comment (StefChatz) [reply src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx]: we are already in a useMemo hook so we can't call this

Comment (linkielink) [reply src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx]: Then move it outside of the useMemo

**Review by linkielink — APPROVED** [review]

File: src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx [hunk 12]
```diff
@@ -28,86 +30,175 @@ export default function useAccountBalancesColumns(
...
+      </Tooltip>
+    )
+  }, [])
+
+  const applyRowStyling = useCallback(
+    (children: React.ReactNode, isWhitelisted: boolean) => (
+      <div
+        className={classNames(
+          'flex items-center',
+          !isWhitelisted && 'opacity-70 hover:opacity-100',
```
Comment (linkielink) [inline src/components/account/AccountBalancesTable/Columns/useAccountBalancesColumns.tsx:56]: Could you please work with a `group-hover` but with individual columns hover here?
Maybe it makes more sense to move the styling and hover to the Row level. The current implementation doesn't feel like you're hovering the whole row but is pretty column specific.

**Review by linkielink — CHANGES_REQUESTED** [review]

File: src/components/common/Table/Row.tsx [hunk 13]
```diff
@@ -43,6 +46,7 @@ export default function Row<T>(props: Props<T>) {
           canExpand && row.getIsExpanded()
             ? 'is-expanded border-t gradient-header'
             : 'hover:bg-white/5',
+          'group',
```
Comment (linkielink) [inline src/components/common/Table/Row.tsx:49]:
```suggestion
          'group/assetRow',
```

File: src/components/common/Table/Row.tsx [hunk 14]
```diff
@@ -71,8 +75,12 @@ export default function Row<T>(props: Props<T>) {
                   type !== 'strategies' &&
                   LEFT_ALIGNED_ROWS.includes(cell.column.id) &&
                   'border-l',
-                type && type !== 'strategies' && getBorderColor(type, cell.row.original as any),
+                type &&
+                  type !== 'strategies' &&
+                  getBorderColor(type, cell.row.original as any, isWhitelisted),
                 cell.column.columnDef.meta?.className,
+                !isWhitelisted && 'opacity-70',
+                !isWhitelisted && 'group-hover:opacity-100',
```
Comment (linkielink) [inline src/components/common/Table/Row.tsx:83]:
```suggestion
                !isWhitelisted && 'group-hover/assetRow:opacity-100',
```

File: src/components/common/Table/Row.tsx [hunk 15]
```diff
@@ -71,8 +75,12 @@ export default function Row<T>(props: Props<T>) {
                   type !== 'strategies' &&
                   LEFT_ALIGNED_ROWS.includes(cell.column.id) &&
                   'border-l',
-                type && type !== 'strategies' && getBorderColor(type, cell.row.original as any),
+                type &&
+                  type !== 'strategies' &&
+                  getBorderColor(type, cell.row.original as any, isWhitelisted),
                 cell.column.columnDef.meta?.className,
+                !isWhitelisted && 'opacity-70',
```
Comment (linkielink) [inline src/components/common/Table/Row.tsx:82]:
```suggestion
                !isWhitelisted && 'opacity-60',
```

**Review by linkielink — CHANGES_REQUESTED** [review]
Please use the specified group to avoid issues when others are using `group` as well.

File: src/components/common/Table/Row.tsx (hunk 15 above)
Comment (linkielink) [reply src/components/common/Table/Row.tsx:82]: I feel it looks a bit better if the opacity is lower than `0.7`

**Review by linkielink — APPROVED** [review]

**Pull request merged**